common/logging/src/lib.rs

```diff
@@ -220,6 +221,25 @@ impl TimeLatch {
 }
 
 pub fn create_tracing_layer(base_tracing_log_path: PathBuf) {
+    let mut tracing_log_path = PathBuf::new();
+
+    // Ensure that `tracing_log_path` only contains directories.
+    for p in base_tracing_log_path.iter() {
+        match metadata(p) {
+            Ok(metadata) => {
+                if metadata.is_dir() {
+                    tracing_log_path = tracing_log_path.join(p);
+                }
+            }
+            // An error here just means that part of `base_tracing_log_path` doesn't exist,
+            // we can safely continue.
+            Err(_) => {
+                tracing_log_path = base_tracing_log_path;
+                break;
+            }
+        }
+    }
```

Comment:
I noticed that the `for` loop unexpectedly breaks in the `Err(_)` statement when running Lighthouse with the `--logfile /Users/akihito/logs/beacon.log` parameter. This happens because the strings passed to `match metadata(p)` are [`/`, `Users`, `akihito`, `logs`, `beacon.log`], and the second string, `Users` (not `/Users`), doesn't exist in my working directory.

I think we can fix this by making the following change:

```diff
    for p in base_tracing_log_path.iter() {
+       tracing_log_path = tracing_log_path.join(p);
+       match tracing_log_path.metadata() {
            Ok(metadata) => {
+               if !metadata.is_dir() {
+                   tracing_log_path.pop();
+                   break;
                }
            }
```

Reply:
thanks for catching that issue! i've pushed up a fix here: [f97e0d4](https://github.com/sigp/lighthouse/pull/6266/commits/f97e0d442ff7647586a6705cd8ab698b5feeb602)

havent had a chance to test it yet, will get to that soon